gobblin-compaction/src/main/java/gobblin/compaction/Dataset.java

```diff
@@ -237,10 +237,22 @@ public void setJobProps(State jobProps) {
     return this.throwables;
   }
 
+  /**
+   * Record a {@link Throwable} in a {@link Dataset}.
+   */
   public synchronized void addThrowable(Throwable t) {
     this.throwables.add(t);
   }
 
+  /**
+   * Skip the {@link Dataset} by setting its {@link DatasetState} to {@link DatasetState#COMPACTION_COMPLETE},
+   * and record the given {@link Throwable} in the {@link Dataset}.
+   */
+  public void skip(Throwable t) {
```

Comment:
Seems you use `synchronized` in `addThrowable`, but not here. If you do need synchronization, I suggest you change `this.throwables` to a `ConcurrentHashSet` or if you really need a `List`, to something like `Collections.synchronizedList`.